Ginger/GingerCore/ALM/OctaneCore.cs

```diff
@@ -226,5 +365,520 @@ public override bool IsServerConnected()
+                {
+                    testCollection.Add(new QCTestInstance() { Id = item.Test.Id, TestId = item.Test.Id, Name = item.Test.Name , CycleId = testSetID });
+                }
+                return testCollection;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to get test instances of test set with REST API", ex);
+                return null;
+            }
+        }
+
+        public QCTestCase GetTestCases(string testcaseID)
+        {
+            LogicalQueryPhrase qd = new LogicalQueryPhrase("id", testcaseID, ComparisonOperator.Equal);
+            IList<IQueryPhrase> filter = new List<IQueryPhrase> { qd };
+            List<Test> test = octaneRepository.GetEntities<Test>(GetLoginDTO(), filter);
+            if (test.Any())
+            {
+                Test testTemp = test.Where(f => true).FirstOrDefault();
+                return new QCTestCase() { Id = testcaseID, TestId = testcaseID, Name = testTemp.Name };
+            }
+            return null;
+        }
+
+        public QCTestCaseStepsColl GetListTSTestSteps(QCTestCase testcase)
+        {
+            char[] lineBreaks = new char[] { '\n' };
+            QCTestCaseStepsColl stepsColl = new QCTestCaseStepsColl();
+            string steps = Task.Run(() => { return octaneRepository.GetTestCaseStep(GetLoginDTO(), testcase.Id); }).Result;
+            int i = 1;
+            if (!string.IsNullOrEmpty(steps))
+            {
+                foreach (string step in steps.Split(lineBreaks))
+                {
+                    stepsColl.Add(new QCTestCaseStep() { Id = Convert.ToString(i), TestId = testcase.Id, Description = step, Name = step, StepOrder = Convert.ToString(i++) });
+                }
+            }
+            return stepsColl;
+        }
+
+        private List<string> GetTCParameterList(string steps)
+        {
+            List<string> parameters = new List<string>();
+            string hasParams = @"(<[a-zA-Z0-9,+/-_()~!@#$%^&*=]+>)";
+            MatchCollection mc = Regex.Matches(steps, hasParams);
+            if (mc != null && mc.Count > 0)
+            {
+                foreach (Match m in mc)
+                {
+                    parameters.Add(m.Value.Substring(1, m.Value.Length - 2));
+                }
+            }
+            return parameters;
+        }
+
+        private void CheckForParameter(QC.QCTSTest newTSTest, string steps)
+        {
+            string hasParams = @"(<[a-zA-Z0-9,+/-_()~!@#$%^&*=]+>)";
+            MatchCollection mc = Regex.Matches(steps, hasParams);
+            if (mc != null && mc.Count > 0)
+            {
+                foreach (Match m in mc)
+                {
+                    QC.QCTSTestParameter newtsVar = new QC.QCTSTestParameter();
+                    newtsVar.Name = m.Value.Substring(1, m.Value.Length - 2);
+                    //if (paramValue != null) { newtsVar.Value = ; }                    
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6317725)